waspc/data/Generator/templates/sdk/server/dbClient.ts

```diff
@@ -1,11 +1,21 @@
+{{={= =}=}}
+{=# areThereAnyEntitiesDefined =}
 import Prisma from '@prisma/client'
 
-
-const createDbClient = () => {
-  const prisma = new Prisma.PrismaClient()
-
-  return prisma
+function createDbClient(): Prisma.PrismaClient {
+  return new Prisma.PrismaClient()
+}
+{=/ areThereAnyEntitiesDefined =}
+{=^ areThereAnyEntitiesDefined =}
+// * Prisma will not generate a PrismaClient if there no
+//   entities in the schema. Trying to init the PrismaClient
+//   will throw an error.
+// * To avoid throwing an error, we return null if there are no
+//   entities in the schema.
+function createDbClient(): null {
+  return null
 }
```

Comment:
This is all great, the PR, I am just wondering what is the best DX.

So they will get `dbClient` which is null. If they try to use it, they will get TS errors. Will they understand why they are getting them? I don't think so, right? They will just be confused that it is `null`, when it shouldn't be.

I wonder if we can somehow communicate to them what is the issue?

Btw, if they are not using TS, but JS, they will just get runtime error, right?

I guess the improvement is that they will be hitting these issues only if they try to use it, instead of Wasp app dying in any case, but still, would be nice to handle this the best we can.

I wonder if we should just not export `dbClient` for them at all. So they can't import it. At least they will be like "ok, for some reason it doesn't exist", which is semantically correct.

Not sure. Leaving it as you did here is also not terrible, I am just trying to think of some potentially nicer ideas DX-wise. Let me know what you think.

Reply:
> So they will get dbClient which is null. If they try to use it, they will get TS errors. Will they understand why they are getting them? I don't think so, right? They will just be confused that it is null, when it shouldn't be.

I guess they won't try to use it if they don't have entities defined? e.g. they would need to do `prisma.something` where `something` is one of entities they defined. 

> I wonder if we should just not export dbClient for them at all. So they can't import it. At least they will be like "ok, for some reason it doesn't exist", which is semantically correct.

We are reexporting the `prisma` symbol from `wasp/server`, so we would need to conditionally render that file now, which is okay. That way we wouldn't import stuff from `dbClient`. We would should conditionally generate the `dbClient.ts`, which is also okay. 

So the option is between:
1. export `null` if not defined _(currently in this PR)_
2. conditionally export from`wasp/server` and generate `dbClient.ts`

The second option is a bit more complex, but not by that much. What do you prefer?

Reply:
Hm you are right, they are not likely to hit it. Unless they want to do some raw queries with prisma or something, but that is not very likely.

If we had infinite time I would probably vote for option (2), but I think it is just not worth it at the moment, the benefits are not super clear -> let's then stick with what we have in this PR!